code/modules/projectiles/guns/projectile/revolver.dm

```diff
@@ -363,26 +363,56 @@
 	else
 		return ..()
 
+
+
 /obj/item/gun/projectile/revolver/doublebarrel/sawoff(mob/user)
 	. = ..()
 	weapon_weight = WEAPON_MEDIUM
 
 /obj/item/gun/projectile/revolver/doublebarrel/attack_self(mob/living/user)
 	var/num_unloaded = 0
+	var/obj/item/storage/belt/bandolier/B
+	for(var/obj/item/I in user.contents)
+		if(istype(I, /obj/item/storage/belt/bandolier))
+			B = I
+			break
+
+	var/shells_to_load = 0 // Start with 0 and increment based on spent shells
+
 	while(get_ammo() > 0)
 		var/obj/item/ammo_casing/CB
 		CB = magazine.get_round(0)
 		chambered = null
 		CB.loc = get_turf(loc)
+		CB.maptext = ""  // Reset the maptext property
```

Comment:
You probably shouldnt be changing the maptext here, but inside of retrieve_item_of_type instead imo